Microsoft.Toolkit.Uwp.UI.Controls/FlexPanel/TypeConverters/FlexAlignItemsTypeConverter.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See the LICENSE file in the project root
+// for the license information.
+//
+// Author(s):
+//  - Laurent Sansonetti (native Xamarin flex https://github.com/xamarin/flex)
+//  - Stephane Delcroix (.NET port)
+//  - Ben Askren (UWP/Uno port)
+//
+using System;
+using System.ComponentModel;
+using System.Globalization;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// String to FlexAlignItems type converter
+    /// </summary>
+    internal class FlexAlignItemsTypeConverter : TypeConverter
+    {
+        public override bool CanConvertFrom(ITypeDescriptorContext context, Type sourceType)
+        {
+            if (sourceType == typeof(string))
+            {
+                return true;
+            }
+
+            return base.CanConvertFrom(context, sourceType);
+        }
+
+        public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo culture, object value)
+        {
+            if (value is string stringValue)
+            {
+                if (Enum.TryParse(stringValue, true, out FlexAlignItems alignitems))
+                {
+                    return alignitems;
+                }
+
+                if (stringValue.Equals("flex-start", StringComparison.OrdinalIgnoreCase))
+                {
+                    return FlexAlignItems.Start;
+                }
+
+                if (stringValue.Equals("flex-end", StringComparison.OrdinalIgnoreCase))
+                {
+                    return FlexAlignItems.End;
+                }
+            }
+
+            throw new InvalidOperationException(string.Format("Cannot convert \"{0}\" into {1}", value, typeof(FlexAlignItems)));
```

Comment:
The codegen here could be improved by using a non-inlined throw helper, so that the string interpolation bit would be moved out of the main body as well. Something like this:

```suggestion
            static object Throw(object value)
            {
                throw new InvalidOperationException(string.Format("Cannot convert \"{0}\" into {1}", value, typeof(FlexAlignItems)));
            }
            
            return Throw(value);
```

This same concept can be applied to all the other similar converters below as well 😊

Reply:
Could you point me to a reference that would help me understand the mechanics of why the suggested approach is better?  This is new to me and I would like to understand what is going on here.

Reply:
Sure thing! I've added a detailed explanation about this in the docs for the `ThrowHelper` type (the same concept applies here, as this dedicated method would act the same way): [preview docs here](https://github.com/MicrosoftDocs/WindowsCommunityToolkitDocs/blob/a6b84d5cce5cdfbec7894645026dbe3342394659/docs/developer-tools/ThrowHelper.md). Hope it helps! 😊

Reply:
So, just to be sure I understand, rather than implementing the change as suggested, I could use ThrowHelper instead?

```csharp
    ThrowHelper.ThrowInvalidOperationException(string.Format("Cannot convert \"{0}\" into {1}", value, typeof(FlexAlignContent)));
```

Reply:
Not in this specific case, no. You need to use that auxiliary method I suggested, as otherwise you'd do all the `string` interpolation _before_ the call to `ThrowHelper`, so all that code would end up in the body of the current method. Whereas with my suggestion all that would be moved to the throw method which would not be inlined.

I've setup a code sample for you so you can see the difference, [here](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMp8gdhoG8aGenHjySBgEsAdhgYAVGLgwB9AJKiANmJgAKMRIQBKbry7VexkQDMG6hAwA8DAAx6jJnoefOMACygQA7g1EwfgCCUADmAK74MOIAoghgMAAOGMIQouoAJABEkh4wDABu2Mrh+Ryi2FEQppY6AL4MPngMyjK4DJ7YovYgDBwIdVk6ANz6znVjJpPGxKwMCKNOPBNLDJP0LEJaUjLyud4+ALIwnhAAJpri847Grm7C5pY29jdud268AkLEKFJevpdtK8Pi5piCeJ4Dv5AgwQhEorF4kkUmlMjk8oViqU+hUqjVdA0mu1Wrh2p1unZev1BiMweM6VNVh99gDdIsPisPgzZvN2bwVnUgA===).
Notice how the second version results in **much** smaller codegen, and all the code is moved away in the throw method that also does the string interpolation for the message to use in the exception 🙂